pytorch_lightning/core/lightning.py

```diff
@@ -1757,3 +1700,54 @@ def get_tqdm_dict(self) -> Dict[str, Union[int, str]]:
         rank_zero_warn("`get_tqdm_dict` was renamed to `get_progress_bar_dict` in v0.7.3"
                        " and this method will be removed in v1.0.0", DeprecationWarning)
         return self.get_progress_bar_dict()
+
+    def _auto_register_hparams(self):
+        """
+        Removes the need to pass in hparams. Instead, we register every argument in init
+        to the module with some caveats:
+        1. we don't overwrite the property if it already exists
+        2. we also store a module_arguments property for model loading and saving
+        """
+        # two frames back is the init of the child module
+        frame = inspect.currentframe()
+        # todo: this shall be flexible to find all init in the path, recursion?
+        frame_args = frame.f_back.f_back.f_locals
+
+        frame_args = _collect_init_args(frame)
+        init_args = {k: v for k, v in frame_args.items()
+                     if k not in ('args', 'kwargs', 'self', '__class__', 'frame', 'frame_args')}
+        child = _get_latest_child(frame)
+
+        # we'll save hparams automatically (renamed to module_arguments)
+        for arg, val in init_args.items():
+            # don't overwrite something already set
+            if hasattr(child, arg):
+                continue
```

Comment:
I feel like here we should throw an error (or warning?), because the user may not know that a certain attribute exists. 
If we just skip it, the user will try to call self.myhparam somewhere in the module and will get an error anyway but may not understand why.